main.ts

```diff
@@ -20,34 +26,84 @@ const DEFAULT_SETTINGS: PluginSettings = {
   lastUpdated: null,
   autoSync: true,
   highlightSortOldestToNewest: true,
+  syncNotesOnly: false,
   logFile: true,
   logFileName: 'Sync.md',
+  frontMatter: false,
+  frontMatterTemplate: `---
+id: {{ id }}
+updated: {{ updated }}
+title: {{ title }}
+author: {{ author }}
+---
+`,
+  headerTemplate: `
+%%
+ID: {{ id }}
+Updated: {{ updated }}
+%%
+
+![]( {{ cover_image_url }})
+
+# About
+Title: {{ title }}
+Authors: {{ authorStr }}
+Category: #{{ category }}
+{%- if tags %}
+Tags: {{ tags }}
+{%- endif %}
+Number of Highlights: =={{ num_highlights }}==
+Readwise URL: {{ highlights_url }}
+Source URL: {{ source_url }}
+Date: [[{{ updated }}]]
+Last Highlighted: *{{ last_highlight_at }}*
+---
+
+# Highlights 
+`,
+  highlightTemplate: `{{ text }} %% highlight_id: {{ id }} %%

```

Comment:
Loving the template engine implementation.

However, can we please change the default template to include the features previous highlighting format contained? The new highlight output doesn't contain the Kindle URL link, kindle highlight color, or the carat denoted identifier that helps with transclusion:
![highlights2](https://user-images.githubusercontent.com/4033621/149646006-90043d7f-a824-43ac-b9c7-4c271a06c595.png)

![highlights1](https://user-images.githubusercontent.com/4033621/149646005-a0d7306e-c45b-4a87-9b3c-7343f87c3a94.png)

Reply:
Hi @jsonMartin, now the delay's on me. I've implemented the changes as requested. Would you mind having a look again?